--- pdf.orig.js	2016-06-20 13:34:52.370608323 +0200
+++ pdf.js	2016-06-20 13:35:36.670757885 +0200
@@ -826,7 +826,9 @@
   function getL10nData(key, args, fallback) {
     var data = gL10nData[key];
     if (!data) {
-      console.warn('#' + key + ' is undefined.');
+      if (Object.keys(gL10nData).length > 0) {
+        console.warn('#' + key + ' is undefined.');
+      }
       if (!fallback) {
         return null;
       }
@@ -10289,10 +10291,17 @@
 var PAGE_NUMBER_LOADING_INDICATOR = 'visiblePageIsLoading';
 var DISABLE_AUTO_FETCH_LOADING_BAR_TIMEOUT = 5000;
 
-PDFJS.imageResourcesPath = './images/';
-  PDFJS.workerSrc = '../build/pdf.worker.js';
-  PDFJS.cMapUrl = '../web/cmaps/';
-  PDFJS.cMapPacked = true;
+var scriptTagContainer = document.body ||
+                         document.getElementsByTagName('head')[0];
+var pdfjsSrc = scriptTagContainer.lastChild.src;
+
+if (pdfjsSrc) {
+  PDFJS.imageResourcesPath = pdfjsSrc.replace(/pdf\.js$/i, 'images/');
+  PDFJS.workerSrc = pdfjsSrc.replace(/pdf\.js$/i, 'pdf.worker.js');
+  PDFJS.cMapUrl = pdfjsSrc.replace(/pdf\.js$/i, 'cmaps/');
+}
+
+PDFJS.cMapPacked = true;
 
 var mozL10n = document.mozL10n || document.webL10n;
 
@@ -11740,7 +11749,7 @@
      * @returns {number}
      */
     get pagesCount() {
-      return this.pdfDocument.numPages;
+      return this.pdfDocument ? this.pdfDocument.numPages : 0;
     },
 
     /**
@@ -13192,7 +13201,11 @@
           !(container = element.parentNode)) {
         throw new Error('Not enough parameters.');
       } else if (this.overlays[name]) {
-        throw new Error('The overlay is already registered.');
+        if (this.active !== name) {
+          this.unregister(name);
+        } else {
+          throw new Error('The overlay is already registered and active.');
+        }
       }
       this.overlays[name] = { element: element,
                               container: container,
@@ -16485,7 +16498,7 @@
   },
 
   get pagesCount() {
-    return this.pdfDocument.numPages;
+    return this.pdfDocument ? this.pdfDocument.numPages : 0;
   },
 
   set page(val) {
@@ -17021,7 +17034,9 @@
   cleanup: function pdfViewCleanup() {
     this.pdfViewer.cleanup();
     this.pdfThumbnailViewer.cleanup();
-    this.pdfDocument.cleanup();
+    if (this.pdfDocument) {
+      this.pdfDocument.cleanup();
+    }
   },
 
   forceRendering: function pdfViewForceRendering() {
@@ -17462,7 +17477,12 @@
   }
 }
 
-document.addEventListener('DOMContentLoaded', webViewerLoad, true);
+// document.addEventListener('DOMContentLoaded', webViewerLoad, true);
+PDFJS.webViewerLoad = function (src) {
+  if (src) DEFAULT_URL = src;
+
+  webViewerLoad();
+}
 
 document.addEventListener('pagerendered', function (e) {
   var pageNumber = e.detail.pageNumber;
@@ -17585,7 +17605,7 @@
 });
 
 window.addEventListener('hashchange', function webViewerHashchange(evt) {
-  if (PDFViewerApplication.pdfHistory.isHashChangeUnlocked) {
+  if (PDFViewerApplication.pdfHistory && PDFViewerApplication.pdfHistory.isHashChangeUnlocked) {
     var hash = document.location.hash.substring(1);
     if (!hash) {
       return;
@@ -17598,7 +17618,7 @@
   }
 });
 
-window.addEventListener('change', function webViewerChange(evt) {
+/*window.addEventListener('change', function webViewerChange(evt) {
   var files = evt.target.files;
   if (!files || files.length === 0) {
     return;
@@ -17627,7 +17647,7 @@
     setAttribute('hidden', 'true');
   document.getElementById('download').setAttribute('hidden', 'true');
   document.getElementById('secondaryDownload').setAttribute('hidden', 'true');
-}, true);
+}, true);*/
 
 function selectScaleOption(value) {
   var options = document.getElementById('scaleSelect').options;
@@ -17738,6 +17758,9 @@
 }, true);
 
 function handleMouseWheel(evt) {
+  // Ignore mousewheel event if pdfViewer isn't loaded
+  if (!PDFViewerApplication.pdfViewer) return;
+  
   var MOUSE_WHEEL_DELTA_FACTOR = 40;
   var ticks = (evt.type === 'DOMMouseScroll') ? -evt.detail :
               evt.wheelDelta / MOUSE_WHEEL_DELTA_FACTOR;
